components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/APIGatewayManager.java

```diff
@@ -96,8 +96,11 @@ public void publishToGateway(WebApp api, APITemplateBuilder builder, String tena
 					// We need to remove the api from the environment since its
 					// relevant url has been removed.
 					operation ="delete";
-					client.deleteApi(tenantDomain);
-					setSecurevaultProperty(api,tenantDomain,environment,operation);
+					if (client.getNonVersionedWebAppData(tenantDomain) != null) {
+						client.deleteNonVersionedWebApp(tenantDomain);
+					}
+					client.deleteVersionedWebApp(tenantDomain);
+					setSecurevaultProperty(api, tenantDomain, environment, operation);
```

Comment:
`setSecurevaultProperty` method was already there, merely used it here (see 5 lines above)
`api` variable was already there, merely used it here (see 5 lines above)